lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -0,0 +1,20 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyIndexer
+  class Configuration
+    extend T::Sig
+
+    sig { returns(T::Array[String]) }
+    attr_accessor :files_to_index
```

Comment:
Is this meant to be changed by the client?

```suggestion
    attr_reader :files_to_index
```

Reply:
Yes, it needs to be an accessor so that people can do
```ruby
RubyIndexer.configure |config|
  config.files_to_index = ...
end
```

But notice that this PR is only moving the code we currently have. Improvements are coming in the next follow ups.